cloud/azure/deploy/schedule/dapr.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2021 Nitric Technologies Pty Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schedule
+
+import (
+	"strings"
+
+	"github.com/nitrictech/nitric/cloud/azure/deploy/exec"
+	"github.com/pkg/errors"
+	"github.com/pulumi/pulumi-azure-native-sdk/app"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+type ScheduleArgs struct {
+	ResourceGroupName pulumi.StringInput
+	Target            *exec.ContainerApp
+	Environment       *app.ManagedEnvironment
+	Cron              string
+}
+
+type Schedule struct {
+	pulumi.ResourceState
+
+	Name      string
+	Component *app.DaprComponent
+}
+
+func NewDaprCronBindingSchedule(ctx *pulumi.Context, name string, args *ScheduleArgs, opts ...pulumi.ResourceOption) (*Schedule, error) {
+	res := &Schedule{
+		Name: name,
+	}
+	normalizedName := strings.ToLower(strings.ReplaceAll(name, " ", "-"))
```

Comment:
Should this be extracted to a function so name normalization is done consistently?

Reply:
We could look at making this common during the refactor for our current cloud implementations (the behaviour is the same for runtime and deploy time).